### PR TITLE
Dynamically tag resources and add password from secret support

### PR DESCRIPTION
**Pull request by todaywasawesome**
Three changes:
1. Set resource names to work off the release name, this is a best practice that makes charts and their resources follow name patterns. 
2. Add support for pulling the password from a secret, as well as a default password if one isn't set in the values. 
3. Set a namespace parameter in resources

Signed-off-by: Dan Garfield <dan@codefresh.io>

### CODE REVIEW

**Comment by todaywasawesome** [comment]
@Addyvan 👍 ?

**Review by Addyvan — APPROVED** [review]
Very reasonable, thanks for this!

**Comment by Addyvan** [comment]
@todaywasawesome 

Forgot to double back but better late than never! 

Rolled out a new version of the chart with the updates from this PR: https://github.com/Addyvan/valheim-k8s/releases/tag/valheim-k8s-1.2.0

File: chart/templates/persistentvolumeclaim.yml [hunk 1]
```diff
@@ -18,7 +19,8 @@ spec:
 apiVersion: v1
 kind: PersistentVolumeClaim
 metadata:
-  name: valheim-server-base-data
+  name: {{ .Release.Name }}-server-base-data
```
Comment (fbuchmeier) [inline chart/templates/persistentvolumeclaim.yml:22]: This is also a breaking change, even when the Helm release name was `valheim-server` before. In this case, the old PVC would have been named `valheim-server-base-data`  and the new one with the same release name `valheim-server-server-base-data`.

**Pull request merged**